plugins/rest/aws_test.go

```diff
@@ -284,6 +284,85 @@ func TestV4Signing(t *testing.T) {
 	assertEq(req.Header.Get("X-Amz-Security-Token"), "MYAWSSECURITYTOKENGOESHERE", t)
 }
 
+func TestV4SigningForApiGateway(t *testing.T) {
+	ts := ec2CredTestServer{}
+	ts.start()
+	defer ts.stop()
+
+	cs := &awsMetadataCredentialService{
+		RoleName:        "my_iam_role", // not present
+		RegionName:      "us-east-1",
+		credServicePath: ts.server.URL + "/latest/meta-data/iam/security-credentials/",
+		tokenPath:       ts.server.URL + "/latest/api/token"}
+	ts.payload = metadataPayload{
+		AccessKeyID:     "MYAWSACCESSKEYGOESHERE",
+		SecretAccessKey: "MYAWSSECRETACCESSKEYGOESHERE",
+		Code:            "Success",
+		Token:           "MYAWSSECURITYTOKENGOESHERE",
+		Expiration:      time.Now().UTC().Add(time.Minute * 2)}
+	req, _ := http.NewRequest("POST", "https://myrestapi.execute-api.us-east-1.amazonaws.com/prod/logs",
+		strings.NewReader("{ \"payload\": 42 }"))
+	req.Header.Set("Content-Type", "application/json")
+
+	err := signV4(req, "execute-api", cs, time.Unix(1556129697, 0))
+
+	if err != nil {
+		t.Error("unexpected error during signing")
```

Comment:
[nit] We could make this a Fatal,
```suggestion
		t.Fatal("unexpected error during signing")
```
since the assertions below probably don't matter if `err != nil` here.